charts/consul/test/unit/client-daemonset.bats

```diff
@@ -1531,6 +1531,20 @@ rollingUpdate:
   [[ "$output" =~ "global.secretsBackend.vault.consulClientRole must be provided if global.secretsBackend.vault.enabled=true" ]]
 }
 
+@test "client/DaemonSet: fail when vault is enabled with tls but autoencrypt is disabled" {
```

Comment:
nit: nice to group the fail tests at the top of the `vault integration` test section with the other fail tests